app/src/main/java/ca/dal/cs/athletemonitor/athletemonitor/RecordActivity.java

```diff
@@ -286,6 +311,31 @@ public void toggleRecordStatus(View v) {
         }
     }
 
+
+    private static void updateLocationData(final UserLocation userLocation){
+        final String username = userLocation.getUsername();
+        final FirebaseDatabase db = FirebaseDatabase.getInstance();
+        final DatabaseReference userLocationRef
+                = db.getReference("user_locations" );
+
+        userLocationRef.addListenerForSingleValueEvent(new ValueEventListener() {
+            @Override
+            public void onDataChange(DataSnapshot dataSnapshot) {
+                if (!dataSnapshot.hasChild(username)) {
+                    userLocationRef.child(username).setValue(username);
+                }
+
+                DatabaseReference userLocatReference = db.getReference("user_locations/" + username);
```

Comment:
maybe I'm misunderstanding, but here can't you just do `userLocationRef.child(username).setValue(userLocation)` ? You already have a DatabaseReference above, why make a new one?

Reply:
If my above comment is correct, this whole `onDataChange` method can just be a single line `userLocationRef.child(username).setValue(userLocation);`